tests/patmat/i9489.scala

```diff
@@ -1,6 +1,6 @@
 import scala.quoted._
 
-def summonTypedType[T : Type](using QuoteContext): String = '[T] match {
+def summonTypedType[T : Type](using QuoteContext): String = Type[T] match {
   case '[Boolean] => "Boolean"
   case '[Byte] => "Byte"
   case _ => "Other"
```

Comment:
It seems `'[T]` is more symmetric here.

Reply:
Indeed, and that is why it should not be used. The syntactic symmetry is what lead to most conceptual misunderstandings of how Type works.

This constructor should also never be used explicitly and giving it a shorter alias does not help users avoid the explicit use.

It is also uncertain that the `case '[...] ` pattern will be kept. We might remove them and use only a single matching abstraction as we do with `inline match`.

Reply:
If the usage is bad, what about disallowing it in syntax?

Reply:
That is my intent as I explain in #9486. This is a first step to help migration. We can drop the syntax later on once users have adapted. I would prefer to have at least one release with a warning.

Reply:
Might be good to bring this up in the meeting, as it's syntax related.

Reply:
This specific case is a corner case the should be avoided. It should have been written explicitly as

```scala
def summonTypedType[T : Type](using QuoteContext): String = summon[Type[T]] match { ...
// or 
def summonTypedType[T](using tp: Type[T])(using QuoteContext): String = tp match { ...
```